File: src/models/File.js
/**
 * File class represents a file in the file system
 * Migrated from C++ File.h/File.cpp
 */
class File {
  /**
   * Create a new file
   * @param {string} id - Unique identifier for the file
   * @param {string} fileName - Name of the file with extension
   * @param {string} folderId - ID of the parent folder
   */
  constructor(id, fileName, folderId) {
    this.id = id;
    this.folderId = folderId;
    this.content = '';
    
    // Parse file name and extension
    const lastDotIndex = fileName.lastIndexOf('.');
    if (lastDotIndex > -1) {
      this.name = fileName.substring(0, lastDotIndex);
      this.extension = fileName.substring(lastDotIndex + 1);
    } else {
      this.name = fileName;
      this.extension = '';
    }
  }

  /**
   * Set the content of the file
   * @param {string} content - Content to be set
   */
  setContent(content) {
    this.content = content;
  }

  /**
   * Get the content of the file
   * @returns {string} - File content
   */
  getContent() {
    return this.content;
  }

  /**
   * Get the file name with extension
   * @returns {string} - File name with extension
   */
  getFileName() {
    if (this.extension) {
      return `${this.name}.${this.extension}`;
    }
    return this.name;
  }

  /**
   * Get the parent folder ID
   * @returns {string} - Parent folder ID
   */
  getFolderId() {
    return this.folderId;
  }

  /**
   * Get the file ID
   * @returns {string} - File ID
   */
  getId() {
    return this.id;
  }
}

module.exports = File;

File: src/models/File.cpp
// src/models/File.cpp

#include <vector>
#include <string>
#include <map>
#include <iostream>
#include "../../include/models/File.h"
using namespace std;

File::File(string id, string fileName, string folderId) : id(id), folderId(folderId)
{
    string n;
    int ind;
    for (int i = 0; i < fileName.size(); i++)
        if (fileName[i] == '.')
        {
            ind = i;
            break;
        }
    n = fileName.substr(0, ind);
    string ex = fileName.substr(ind + 1, n.size() - ind - 1);
    name = n;
    extension = ex;
}

void File::setContent(string content) { this->content = content; }

string File::getContent() { return content; }

string File::getId() { return id; }

string File::getFileName() { return name + "." + extension; }

string File::getFolderId() { return folderId; }
